--- v4l-utils-1.2.0/utils/v4l2-compliance/Makefile.am.orig	2014-06-29 10:07:42.968450048 +0200
+++ v4l-utils-1.2.0/utils/v4l2-compliance/Makefile.am	2014-06-29 10:18:18.901770128 +0200
@@ -2,6 +2,6 @@
 v4l2_compliance_SOURCES = v4l2-compliance.cpp v4l2-test-debug.cpp v4l2-test-input-output.cpp \
 	v4l2-test-controls.cpp v4l2-test-io-config.cpp v4l2-test-formats.cpp v4l2-test-buffers.cpp \
 	v4l2-test-codecs.cpp v4l2-compliance.h cv4l-helpers.h v4l-helpers.h
-v4l2_compliance_LDADD = ../../lib/libv4l2/libv4l2.la ../../lib/libv4lconvert/libv4lconvert.la
+v4l2_compliance_LDADD = ../../lib/libv4l2/libv4l2.la ../../lib/libv4lconvert/libv4lconvert.la -lpthread
 
 EXTRA_DIST = Android.mk fixme.txt
